test/automated/log_ignored/substitute.rb

```diff
@@ -0,0 +1,40 @@
+require_relative "../automated_init"
+
+context "LogIgnored" do
+  context "Substitute" do
+    message = Controls::Message.example
+    reason = Controls::Reason.example
+
+    substitute = LogIgnored::Substitute.build
+
+    context "When not invoked" do
+      test "called? returns false" do
+        refute substitute.called?
+      end
+    end
+
+    context "When invoked" do
+      substitute.(message, reason)
+    end
+
+    test "called? returns false" do
+      assert substitute.called?
+    end
+
+    test "called_with? returns true for maatching arguments" do
+      assert substitute.called_with?(message, reason)
+    end
+
+    test "called_with? returns false for non matching reason" do
+      other_reason = Controls::Reason::Other.example
+
+      refute substitute.called_with?(message, other_reason)
+    end
+
+    test "called_with? returns false for non matching message" do
+      other_message = Controls::Message::Other.example
+
+      refute substitute.called_with?(other_message, reason)
+    end
+  end
```

Comment:
remove it - it is a copy of above test

Reply:
No it's not. It's checking that it fails when the message is different, not reason. In order to implement this I had to investigate message equality - turns out it's based on attributes not ID.